service/frontend/workflowHandler.go

```diff
@@ -3464,3 +3465,29 @@ func (wh *WorkflowHandler) validateUTF8String(
 	}
 	return nil
 }
+
+func (wh *WorkflowHandler) trimHistoryNode(
+	ctx context.Context,
+	namespaceID string,
+	workflowID string,
+	runID string,
+) {
+	response, err := wh.historyClient.GetMutableState(ctx, &historyservice.GetMutableStateRequest{
+		NamespaceId: namespaceID,
+		Execution: &commonpb.WorkflowExecution{
+			WorkflowId: workflowID,
+			RunId:      runID,
+		},
+	})
+	if err != nil {
+		return // abort
+	}
+
+	// best effort
+	_, _ = wh.persistenceExecutionManager.TrimHistoryBranch(&persistence.TrimHistoryBranchRequest{
```

Comment:
Will the persistence call write a log on error? If not, we need to write a log for error.